.github/workflows/gen-openapi.yml

```diff
@@ -0,0 +1,28 @@
+name: update-api-schema
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review, labeled, unlabeled]
+  push:
+    branches: [main]
+jobs:
+  gen-openapi:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.head_ref }}
+      - uses: actions-rs/cargo@v1
+        with:
+          command: run
+          args: -p utils --bin gen-openapi
+      - uses: actions/upload-artifact@v3
```

Comment:
We commit `openApi` schema to the repo, so I there is no reason to publish it as an artifact